man/man8/zfs.8

```diff
@@ -1627,6 +1747,37 @@ The values of user properties are arbitrary strings, are always inherited, and a
 with the \fBzfs create\fR command set up and enable the swap area using the
 \fBmkswap\fR(8) and \fBswapon\fR(8) commands.  Do not swap to a file on a
 \fBZFS\fR file system. A \fBZFS\fR swap file configuration is not supported.
+.SS "Encryption"
+.LP
+Enabling the \fBencryption\fR feature allows the user to create natively
+encrypted filesystems and volumes. \fBZFS\fR will encrypt all user data
+including file and zvol data, file attributes, ACLs, permission bits, directory
+listings, and FUID mappings. \fBZFS\fR will not encrypt metadata related to the
+pool structure, including dataset names, dataset hierarchy, file size, file
+holes, and dedup tables. Key rotation is managed internally by the \fBZFS\fR
+kernel module and changing the user's key does not require re-encrypting the
+entire dataset. Datasets can be scrubbed, resilvered, renamed, and deleted
+without the encryption keys being loaded (see the \fBkeylocation\fR property
+and \fBzfs load-key\fR subcommand for more info).
+.sp
+.LP
+Encryption changes the behavior of a few \fBZFS\fR operations. Encryption is
+applied after compression so compression ratios are preserved. Normally
+checksums in ZFS are 256 bits long, but for encrypted data the checksum is
+128 bits of the user-chosen checksum and 128 bits of MAC from the encryption
+suite, which provides additional protection against maliciously altered data.
+Deduplication is still possible with encryption enabled but for security,
+datasets will only dedup against themselves, their snapshots, and their clones.
+.sp
+.LP
+There are a few limitations on encrypted datasets. Encrypted data cannot be
+embedded via the \fIembedded_data\fR feature. Encrypted datasets may not
+have \fIcopies=3\fR since the implementation stores some encryption metadata
+where the third copy would normally be. Since compression is applied before
+encryption datasets may be vulnerable to a CRIME-like attack if applications
+accessing the data allow for it. Deduplication with encryption will leak
+information about which blocks are equivalent in a dataset and will incur an
+extra CPU cost per block written.
```

Comment:
The extra CPU overhead comes from running the HMAC of the plaintext to generate the IV and salt. This is in addition to the dedup checksum and the encryption CPU overhead.